# Confidence Interval Estimation for Print-Head Durability

This notebook demonstrates how to construct **99% confidence intervals** for the mean durability of print-heads, tested until failure.  
Because the test is **destructive**, the sample size is small (n = 15).

---


In [1]:
import statistics
import math
from scipy import stats

# Data (durability in millions of characters)
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48,
        1.20, 1.33, 1.18, 1.22, 1.29]

n = len(data)

# Sample statistics
xbar = statistics.mean(data)
s = statistics.stdev(data)

xbar, s, n

(1.2386666666666666, 0.19316412956959933, 15)

## a) 99% Confidence Interval Using Sample Standard Deviation (t-distribution)

- Population standard deviation **unknown**  
- Sample size is **small** (n = 15)  
- Use **t-distribution** with df = n-1  


In [3]:
alpha = 0.01
df = n - 1

# t critical value
t_crit = stats.t.ppf(1 - alpha/2, df=df)

# Standard error and margin of error
se_t = s / math.sqrt(n)
moe_t = t_crit * se_t

# Confidence interval
ci_t = (xbar - moe_t, xbar + moe_t)
t_crit, se_t, moe_t, ci_t

(np.float64(2.976842734370834),
 0.049874763793847324,
 np.float64(0.14846932822817593),
 (np.float64(1.0901973384384906), np.float64(1.3871359948948425)))

## b) 99% Confidence Interval Using Known Population Standard Deviation (z-distribution)

- If σ = 0.2 is **known**, we use the **normal distribution** (z).  


In [4]:
sigma = 0.2

# z critical value
z_crit = stats.norm.ppf(1 - alpha/2)

# Standard error and margin of error
se_z = sigma / math.sqrt(n)
moe_z = z_crit * se_z

# Confidence interval
ci_z = (xbar - moe_z, xbar + moe_z)
z_crit, se_z, moe_z, ci_z

(np.float64(2.5758293035489004),
 0.051639777949432225,
 np.float64(0.13301525327090588),
 (np.float64(1.1056514133957607), np.float64(1.3716819199375725)))

## Conclusion

- **t-based 99% CI** (using sample s): approximately (1.090, 1.387) million characters  
- **z-based 99% CI** (using known σ=0.2): approximately (1.106, 1.372) million characters  

The **t-interval** is slightly wider to reflect extra uncertainty from estimating σ.  
Both intervals suggest the mean durability is around **1.2–1.3 million characters** before failure.  
